src/sentry/models/apikey.py

```diff
@@ -96,9 +102,8 @@ def sanitize_relocation_json(
         sanitizer.set_name(json, SanitizableField(model_name, "label"))
 
 
-def is_api_key_auth(auth: object) -> bool:
+def is_api_key_auth(auth: object) -> TypeGuard[AuthenticatedToken | ApiKey | ApiKeyReplica]:
```

Comment:
Just for my understanding of typeguard, is it valid to use it here when we check a property on the instance too?

Reply:
TypeGuard is basically a custom `isinstance` narrowing for the type checker: https://youtu.be/Zv2K2AVy5lA  (`TypeIs` is generally a better `TypeGuard` but wasn't necessary in this particular situation)

basically in the True branch of `if is_api_key_auth(o):`  then it is known to the type checker that `o` must be of type `AuthenticatedToken | ApiKey | ApiKeyReplica`

in reality our `request.auth` is `AuthenticatedToken | None` and so these functions will be sort of useless once I teach the type system that in a followup PR

Reply:
Hmm ok... it seems like it changes the meaning of the function though? Previously this was saying that the auth object uses api key auth, not that it is a specific type. Or is it fine for it to have these dual meanings?

Reply:
TypeGuard decays to `bool` so it's the same thing